src/vm/eetwain.cpp

```diff
@@ -3099,7 +3099,17 @@ size_t EECodeManager::GetCallerSp( PREGDISPLAY  pRD )
     {
         EnsureCallerContextIsValid(pRD, NULL);
     }
-    return (size_t) (GetSP(pRD->pCallerContext));
+
+    size_t res = GetSP(pRD->pCallerContext);
+
+#ifdef _TARGET_X86_
+    EECodeInfo codeInfo;
+    codeInfo.Init((PCODE) pRD->ControlPC);
+
+    res -= codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
```

Comment:
What if we stored the callerSP in the REGDISPLAY in the OOPStackUnwinderX86::VirtualUnwind when we set the EstablisherFrame? Then we would not have to initialize the EECodeInfo and call the GetStackParameterSize here again.